mlflow/pipelines/pipeline.py

```diff
@@ -140,7 +143,7 @@ def clean(self, step: str = None) -> None:
 
     def _get_step(self, step_name) -> BaseStep:
         """Returns a step class object from the pipeline."""
-        steps = self._steps or self._resolve_pipeline_steps()
```

Comment:
What's the reason to remove `self._resolve_pipeline_steps()`? Does it cause any harm?

Reply:
(This is related to the first item in my PR description) `self._resolve_pipeline_steps()` will return new `BaseStep` instances. This line means `self._steps` may contain stale information so `_get_step()` also tries to get the step info from the config file. This PR changes the logic to make `self._steps` always up-to-date (by reloading before each pipeline action) so we don't reload elsewhere (here).

Reply:
Got it. Leaving the code as is should not cause any harm either, since `self._steps` is always populated at this point. But it could be a stopgap in the future if one misses to update the steps.